File: problems/2053-kth-distinct-string-in-an-array/solution.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        counts = {}
        for s in arr:
            counts[s] = counts.get(s, 0) + 1
        
        distinct_strings = []
        for s in arr:
            if counts[s] == 1:
                distinct_strings.append(s)
        
        if len(distinct_strings) < k:
            return ""
        else:
            return distinct_strings[k - 1]

File: problems/2053-kth-distinct-string-in-an-array/solution.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> counts;
        for (const string& s : arr) {
            counts[s]++;
        }

        vector<string> distinct_strings;
        for (const string& s : arr) {
            if (counts[s] == 1) {
                distinct_strings.push_back(s);
            }
        }

        if (distinct_strings.size() < k) {
            return "";
        } else {
            return distinct_strings[k - 1];
        }
    }
};